main.py

```diff
@@ -347,109 +353,249 @@ async def score(self, interaction: discord.Interaction, player: str):
+
+            field_count += 1
+            boss_icon = find_emoji(self._discord_client.emojis, boss["emoji_key"])
+            working_embed.add_field(
+                name=f"{boss_icon} {boss['points']:,} points",
+                value=f"{EMPTY_SPACE}{boss['kc']:,} kc",
+            )
+
+        boss_embeds.append(working_embed)
+        boss_page_count = len(boss_embeds)
+
+        for index, embed in enumerate(boss_embeds):
+            embed.title = f"{rank_icon} {member.display_name} | Bossing Points"
+            embed.description = (
+                f"Breakdown of **{boss_point_counter:,}** points awarded for boss kc."
+            )
+
+            if boss_page_count > 1:
+                embed.title = "".join(embed.title) + f" ({index+1}/{boss_page_count})"
+
+            if index + 1 == boss_page_count:
+                if len(embed.fields) % 3 != 0:
+                    embed.add_field(name="", value="")
+
+        raid_breakdown_embed = build_response_embed(
+            f"{rank_icon} {member.display_name} | Raid Points",
+            "",
+            rank_color,
         )
 
-        with open(path, "rb") as f:
-            discord_file = discord.File(f, filename="breakdown.txt")
-            await interaction.followup.send(
-                f"Total Points for {player}: {total_points} {rank_icon}",
-                file=discord_file,
+        raid_point_counter = 0
+        for raid in raids:
+            raid_point_counter += raid["points"]
+            raid_icon = find_emoji(self._discord_client.emojis, raid["emoji_key"])
+            raid_breakdown_embed.add_field(
+                name=f"{raid_icon} {raid['points']:,} points",
+                value=f"{EMPTY_SPACE}{raid['kc']:,} kc",
+            )
+
+        raid_breakdown_embed.description = f"Breakdown of **{raid_point_counter:,}** points awarded for raid completions."
+
+        clue_breakdown_embed = build_response_embed(
+            f"{rank_icon} {member.display_name} | Cluescroll Points",
+            "Points awarded for cluescroll completions.",
+            rank_color,
+        )
+
+        clue_point_counter = 0
+        clue_icon = find_emoji(self._discord_client.emojis, "cluescroll")
+        for clue in clues:
+            clue_point_counter += clue["points"]
+            clue_breakdown_embed.add_field(
+                name=f"{clue_icon} {clue['points']:,} points",
+                value=f"{EMPTY_SPACE}{clue['kc']:,} {clue.get("display_name", clue['name'])}",
             )
 
-    async def ingots(self, interaction: discord.Interaction, player: str):
-        """View ingots for a Runescape playername.
+        clue_breakdown_embed.description = f"Breakdown of **{clue_point_counter:,}** points awarded for cluescroll completions."
+
+        menu = ViewMenu(
+            interaction,
+            menu_type=ViewMenu.TypeEmbed,
+            show_page_director=True,
+            timeout=600,
+            delete_on_timeout=True,
```

Comment:
Just for my education: we hold "the menu" with all pages for 10 minutes, then reaction emotes are deleted so user can't use next/prev pages? How do we free memory from those pages? Is it some coroutine inside menu.start()?

Reply:
So I looked through the documentation for this library and wasn't able to answer your question. I have linked the repo below. I have tested this locally by spawning 20+ menus and watching the memory footprint return to normal once they have been removed.

To be clear, we indeed hold the menu and its pages for 10 mins. After the 10 mins is up the entire message is removed. I thought this was cleaner. However if sentiment changes it is a quick config change.

https://github.com/Defxult/reactionmenu